logstash-core/spec/logstash/config/mixin_spec.rb

```diff
@@ -164,36 +164,74 @@
       end
     end
     
-    ENV["MIXIN_SPEC_ENV_VAR"] = "123"
-
-    subject {
-      plugin_class.new({
-        "oneString" => "${notExistingVar}",
-        "oneBoolean" => "${notExistingVar:true}",
-        "oneNumber" => "${MIXIN_SPEC_ENV_VAR}",
-        "oneArray" => [ "first array value", "$MIXIN_SPEC_ENV_VAR" ],
-        "oneHash" => { "key" => "$MIXIN_SPEC_ENV_VAR" }
-      })
-    }
-     
-    it "should have oneString param as empty string (env var not found)" do
-      expect(subject.oneString).to(be == "")
+    before do
+      ENV["MIXIN_SPEC_ENV_VAR"] = "123"
     end
 
-    it "should have oneNumber param with environment variable injected" do
-      expect(subject.oneNumber).to(be == 123)
+    after do
+      ENV.delete("MIXIN_SPEC_ENV_VAR")
     end
 
-    it "should have oneBoolean param with default value" do
-      expect(subject.oneBoolean).to(be == true)
-    end
+    context "when an environment variable is not set" do
+      context "and no default is given" do
+        before do
+          # Canary. Just in case somehow this is set.
+          expect(ENV["NoSuchVariable"]).to be_nil
+        end
+
+        it "should raise a configuration error" do
+          expect do
+            plugin_class.new("example" => "${NoSuchVariable}")
```

Comment:
You set here option `example`, while this option is not defined in `subject`.
Could be better to set `plugin_class.new("oneString" => "${NoSuchVariable}")` ?